include/pybind11/stl.h

```diff
@@ -305,6 +314,59 @@ template<> struct type_caster<std::experimental::nullopt_t>
     : public void_caster<std::experimental::nullopt_t> {};
 #endif
 
+#if defined(PYBIND11_HAS_FILESYSTEM)
+template<> struct type_caster<std::filesystem::path> {
+
+    template<typename T> static PyObject* unicode_from_fs_native(T w) {
+        if constexpr (std::is_same_v<T, std::string>) {
+            return PyUnicode_DecodeFSDefaultAndSize(w.c_str(), ssize_t(w.size()));
+        } else if constexpr (std::is_same_v<T, std::wstring>) {
+            return PyUnicode_FromWideChar(w.c_str(), ssize_t(w.size()));
+        }
+    }
+
+    static handle cast(std::filesystem::path path, return_value_policy, handle) {
+        PyObject* py_str = unicode_from_fs_native(path.native());
```

Comment:
You took `path` by value. Now that you've paid the price of the copy, use `std::move(path).native()`. (Hopefully in C++23) it will let you move the string out.

Maybe we need two overloads - `const&` and `&&`?

Reply:
good point, but I think just taking `const&` is enough here?  changed accordingly.